github/pulls.go

```diff
@@ -253,12 +253,16 @@ type PullRequestMergeResult struct {
 
 // PullRequestOptions lets you define how a pull request will be merged.
 type PullRequestOptions struct {
-	Squash bool
```

Comment:
by removing this any user of our library who is doing
```
ops := &PullRequestOptions {
    squash: true,
}
```
will be unable to compile. We "could" keep supporting them by automatically turning `Squash` into a `MergeMethod = "squash"` if we so chose. We don't have to. If github breaks the API I don't know if we really need to be forward compatible with our users....